pkgs/development/python-modules/leanclient/default.nix

```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  poetry-core,
+  orjson,
+  tqdm,
+}:
+
+buildPythonPackage rec {
+  pname = "leanclient";
+  version = "0.1.13";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-cSow9A4AnKnv3y0HUkeM28FYRsgCBTRKVpWDpCH1juc=";
+  };
+
+  build-system = [ poetry-core ];
+
+  dependencies = [
+    orjson
+    tqdm
+  ];
+
+  doCheck = true;
```

Comment:
You're probably missing something here like pytest hooks ? Check other python derivations to get some inspiration!